pallets/pallet-web3-names/src/lib.rs

```diff
@@ -314,7 +314,11 @@ pub mod pallet {
 			let (decoded_name, is_claimed) = Self::check_banning_preconditions(name)?;
 
 			if is_claimed {
-				Self::unregister_name(&decoded_name)?;
+				let owner = Self::unregister_name(&decoded_name)?.owner;
+				Self::deposit_event(Event::<T>::Web3NameReleased {
```

Comment:
Why not having the event genereated in the `unregister_name` function?

Reply:
Yeah that is a good idea! I also moved the claimed event. [52c8b11](https://github.com/KILTprotocol/kilt-node/pull/586/commits/52c8b1138ea5c4cbdc928912172921d8a41b100e)